BookReader/BookReader.js

```diff
@@ -6056,60 +4533,103 @@ BookReader.prototype.getNavHeight = function(ignoreDisplay) {
   return 0;
 }
 
+//------------------------------------------------------------------------------
+// Basic Usage built-in Methods (can be overridden through options)
 
-/**
- * Get's the page resume value, for remembering reader's page
- * Can be overriden for different implementation
- * @return {Number|null}
- */
-BookReader.prototype.getResumeValue = function() {
-    var val = BookReader.docCookies.getItem('br-resume');
-    if (val !== null) return parseInt(val);
-    else return null;
-}
+
+// This implementation uses options.data value for populating BookReader
+BookReader.prototype.getNumLeafs = function() {
+   return this._getDataFlattend().length;
+},
+BookReader.prototype.getPageWidth = function(index) {
+    if (index < 0 || index >= this._getDataFlattend().length) return;
+    return this._getDataFlattend()[index].width;
+};
+BookReader.prototype.getPageHeight = function(index) {
+    if (index < 0 || index >= this._getDataFlattend().length) return;
+    return this._getDataFlattend()[index].height;
```

Comment:
Suggestion: Move common code for getting the page at an index into a private method.

Reply:
Done